.github/workflows/tests.yml

```diff
@@ -10,20 +10,45 @@ on:
 
 # This allows a subsequently queued workflow run to interrupt previous runs.
 concurrency:
-  group: "${{ github.workflow }} @ ${{ github.ref }}"
+  group: '${{ github.workflow }} @ ${{ github.ref }}'
   cancel-in-progress: true
 
 jobs:
-  # Since the name of the matrix job depends on the version, we define another job with a more stable name.
-  test_results:
-    if: ${{ always() }}
+  orchestrate:
+    name: Orchestrate matrix from Docker Hub
     runs-on: ubuntu-latest
-    name: Test Results
-    needs: [test]
+    outputs:
+      postgis_tags: ${{ steps.prepare.outputs.postgis_tags }}
     steps:
-      - run: '[[ "${{ needs.test.result }}" == "success" ]]'
+      - name: Query Docker Hub tags and build matrix
+        id: prepare
+        run: |
+          # https://www.postgresql.org/support/versioning/
+          pg_targets='[14, 15, 16, 17, 18]'
+
+          tags=$(
+            curl -s "https://hub.docker.com/v2/repositories/postgis/postgis/tags?page_size=100" |
+            jq --compact-output '
+              .results
+              | map(.name) as $tags
+              | '"$pg_targets"'
+              | map(
+                . as $target
+                | $tags
+                | map(select(test("^\($target)-\\d+\\.\\d+$")))[0]
+              )
+            '
+          )
+
+          if [[ $(jq 'length' <<< "$tags") -ne $(jq 'length' <<< "$tags") ]]; then
+            echo "Error: Could not find tags for all PostgreSQL targets" >&2
+            exit 1
+          fi
+
+          echo "postgis_tags=$tags" >> $GITHUB_OUTPUT
```

Comment:
what's the reasoning behind needing to do this instead of an array of versions? is it just because postgis does not have the same versions as postgres in docker?